optaplanner-core/src/main/java/org/optaplanner/core/impl/partitionedsearch/DefaultPartitionedSearchPhase.java

```diff
@@ -176,13 +177,12 @@ private ExecutorService createThreadPoolExecutor(int partCount) {
         Termination<Solution_> partTermination = new OrCompositeTermination<>(childThreadPlumbingTermination,
                 termination.createChildThreadTermination(solverScope, ChildThreadType.PART_THREAD));
         List<Phase<Solution_>> phaseList = new ArrayList<>(phaseConfigList.size());
-        int partPhaseIndex = 0;
+        PhaseCounter<Solution_> partPhaseCounter = new PhaseCounter<>(); // Phase count independent of the outer solver.
         for (PhaseConfig phaseConfig : phaseConfigList) {
```

Comment:
When you're inside the partitioned search, the phase index starts from 0 again.
I am keeping this behavior.

Reply:
Yes, because the logging with show that they are on another thread. (each log line also includes the thread name) => context

Partition threads each need to have a separate phase counter because they run in parallel. So they all start from zero.